code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -49,7 +49,6 @@
 	R.speed = 0 // Remove upgrades.
 	R.ionpulse = FALSE
 	R.magpulse = FALSE
-	R.weather_immunities = initial(R.weather_immunities)

```

Comment:
Oh, that is understandable. I will just avoid doing that, then. It is intended that silicons are ashproof, it is defined that way silicon.dm. 

This line was originally intended for resetting ash protection on mining borgs, as only they had it. It was later given to all silicons by another coder, and instead this line was used to remove lava protection from upgraded mining borgs.